### PR TITLE
`Development`: Add latest version to database for better migration management in the future

### PR DESCRIPTION
**Pull request by krusche**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
We want to verify migration paths in the future. Therefore, we need to know which "latest" version was executed before the current "new" version is started

### Description
Store the latest version in the database and store it with native SQL statements

Note: I know the code quality is not the highest level, but it should be fine for now (except you find some really problematic aspects)

### Steps for testing
Start Artemis locally in this branch and check if a new database table with the latest version is created.
No SQL errors or startup errors should occur in the logs

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java [hunk 1]
```diff
@@ -64,4 +89,52 @@ public SpringLiquibase liquibase(@LiquibaseDataSource ObjectProvider<DataSource>
...
+     * @param event used to prevent this method for running in tests
+     */
+    @EventListener()
+    public void storeCurrentVersionToDatabase(ApplicationReadyEvent event) {
+        if (event.getApplicationContext().getEnvironment().acceptsProfiles(Profiles.of(SPRING_PROFILE_TEST))) {
+            return;
+        }
+        try (var statement = createStatement()) {
+            if (previousVersionString == null) {
+                log.info("Insert latest version " + currentVersionString + " into database");
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java:126]:
```suggestion
                log.info("Insert latest version {} into database", currentVersionString);
```

File: src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java [hunk 2]
```diff
@@ -64,4 +89,52 @@ public SpringLiquibase liquibase(@LiquibaseDataSource ObjectProvider<DataSource>
...
+                statement.executeUpdate("INSERT INTO artemis_version (latest_version) VALUES('" + currentVersionString + "');");
+            }
+            else {
+                log.info("Update latest version to " + currentVersionString + " in database");
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java:130]:
```suggestion
                log.info("Update latest version to {} in database", currentVersionString);
```

File: src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java [hunk 3]
```diff
@@ -24,24 +32,41 @@ public class LiquibaseConfiguration {
...
-        SpringLiquibase liquibase = SpringLiquibaseUtil.createSpringLiquibase(liquibaseDataSource.getIfAvailable(), liquibaseProperties, dataSource.getIfUnique(),
+            ObjectProvider<DataSource> dataSourceObjectProvider, DataSourceProperties dataSourceProperties) {
+        SpringLiquibase liquibase = SpringLiquibaseUtil.createSpringLiquibase(liquibaseDataSource.getIfAvailable(), liquibaseProperties, dataSourceObjectProvider.getIfUnique(),
                 dataSourceProperties);
+
+        if (!env.acceptsProfiles(Profiles.of(SPRING_PROFILE_TEST))) {
+            dataSource = dataSourceObjectProvider.getIfUnique();
+            currentVersionString = buildProperties.getVersion();
+            previousVersionString = getPreviousVersionElseThrow();
+            log.info("The previous version was " + previousVersionString);
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java:67]:
```suggestion
            log.info("The previous version was {}", previousVersionString);
```

File: src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java [hunk 4]
```diff
@@ -64,4 +89,52 @@ public SpringLiquibase liquibase(@LiquibaseDataSource ObjectProvider<DataSource>
...
+        return connection.createStatement();
+    }
+
+    /**
+     * Stores the current version in the database after the application is ready
+     * @param event used to prevent this method for running in tests
+     */
+    @EventListener()
+    public void storeCurrentVersionToDatabase(ApplicationReadyEvent event) {
+        if (event.getApplicationContext().getEnvironment().acceptsProfiles(Profiles.of(SPRING_PROFILE_TEST))) {
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java:121]: Same as line 63; Can you move this into a method?

**Review by Strohgelaender — CHANGES_REQUESTED** [review]

File: src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java (hunk 4 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java]: done

**Review by Strohgelaender — APPROVED** [review]
code

**Pull request merged**